File: libs/storage/src/main/java/com/example/iter/common/storage/S3StorageProperties.java
package com.example.iter.common.storage;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;
import org.springframework.validation.annotation.Validated;

import java.time.Duration;

@Getter
@Setter
@Validated
@Component
@ConfigurationProperties(prefix = "app.storage.s3")
public class S3StorageProperties {

    @NotBlank
    private String bucket;

    @NotBlank
    private String region;

    private String publicBaseUrl;

    @NotBlank
    private String keyPrefix = "equipment";

    private Duration presignedUrlValidity = Duration.ofMinutes(5);
}

File: libs/storage/src/main/kotlin/com/example/iter/common/storage/S3StorageProperties.kt
package com.example.iter.common.storage

import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import org.springframework.validation.annotation.Validated
import java.time.Duration

@Validated
@Component
@ConfigurationProperties(prefix = "app.storage.s3")
class S3StorageProperties {

    @NotBlank
    lateinit var bucket: String

    @NotBlank
    lateinit var region: String

    var publicBaseUrl: String? = null

    @NotBlank
    var keyPrefix: String = "equipment"

    var presignedUrlValidity: Duration = Duration.ofMinutes(5)
}
